profiles/factory.py

```diff
@@ -1,10 +1,36 @@
-from flask import Flask
+from flask import Flask, jsonify, make_response, request
+from profiles.api.oauth2 import OAUTH2_BP
 from profiles.api.ping import PING_BP
+from werkzeug.exceptions import HTTPException, InternalServerError
 
 
-def create_app():
+def create_app(config):
     app = Flask(__name__)
+    app.TRAP_HTTP_EXCEPTIONS = True
+    app.config.update(dict(
+        config=config
+    ))
 
+    app.register_blueprint(OAUTH2_BP, url_prefix='/oauth2')
     app.register_blueprint(PING_BP)
 
+    def http_error_handler(exception):
+        if not isinstance(exception, HTTPException):
+            exception = InternalServerError(getattr(exception, 'message', None))
+
+        if request.path.startswith('/oauth2/') and not request.path.startswith('/oauth2/token'):
```

Comment:
Is there a cleaner way to do this?

Reply:
Some paths are meant to be interacted with a browser, others from applications. Better to split it on those lines rather than `Accept` (looking at the latter from a browser should still return JSON rather than HTML).

Reply:
In fact, it's more complicated than that. We need _two_ types of JSON errors: OAuth2 errors, and out standard API errors (RFC 7807).

Reply:
Make that _four_ types of error in total: HTML, redirect, and 2 JSON types. 😓